File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        d={"I":1,
           "V":5,
           "X":10,
           "L":50,
           "C":100,
           "D":500,
           "M":1000}
        total=d[s[0]]
        for i in range(1,len(s)):
            if d[s[i-1]]<d[s[i]]:
                total+=d[s[i]]-2*d[s[i-1]]
            else:
                total+=d[s[i]]

            print(total)
        return total
            
            

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character,Integer> map= new HashMap<>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);
        
        int total=map.get(s.charAt(0));
        for(int i=1; i<s.length();i++){
            int val=map.get(s.charAt(i));
            int prev=map.get(s.charAt(i-1));
            total+=val;
            if (i>0 && val>prev)
                total-=2*prev;
        }
        return total;
        
    }
}